### DISCUSSION TITLE
Instructions for usage with Vitest don't work for me

### DISCUSSION THREAD

**Discussion by FlinnBurgess** [testing-library/jest-dom#587] [Q&A]
Hello, I've been helping my team switch from Jest to Vitest, and I noticed that I was getting a lot of type errors along the lines of `Property toBeInTheDocument does not exist on type JestMatchers<HTMLElement>`. After a little digging I realised I could fix it by replacing
```
import '@testing-library/jest-dom/vitest';
```
with
```
import '@testing-library/jest-dom'
```
in my `vitest.setup.ts` file.
This seems be the opposite of the instructions on the npm page, so I'm wondering if this is a side effect of me doing something else wrong?

**Comment by sQVe** [comment]
I have the same issue.

**Comment by andykenward** [comment]
Have a read-through of https://github.com/testing-library/jest-dom/issues/567#issuecomment-1881690215 it might help.

**Comment by sQVe** [comment]
@andykenward I already have my `vitest.setup.ts` file included in the `tsconfig.json`. Sadly still not working, unless I use `import '@testing-library/jest-dom'`.

**Comment by sQVe** [comment]
Got it working on my end. I needed to ensure that our monorepo didn't have conflicting `vitest` version.

**Reply by andykenward** [reply]
What package manager are you using for your monorepo? npm / yarn / pnpm ?

**Reply by JanderSilv** [reply]
Hello, I'm facing this issue with yarn workspaces with no conflicted version of vitest or vite. Without hoisting the dependencies works fine. Any ideas?

**Comment by filetvignon** [comment]
I was having the same issue and it wasn't enough to just add the setup file to `tsconfig.json`.  
I only got it working by adding `@testing-library/jest-dom/vitest` to `compilerOptions.types` in `tsconfig.json`:

```json
{
  "compilerOptions": {
    "types": ["@testing-library/jest-dom/vitest"]
  }
}
